src/status_im/contexts/shell/share/wallet/view.cljs

```diff
@@ -74,18 +74,66 @@
             :on-legacy-press     #(reset! wallet-type :wallet-legacy)
             :on-settings-press   #(open-preferences @selected-networks)}]]]))))
 
+(defn- indicator
+  [is-active]
+  [rn/view
+   {:style {:width           8
+            :height          8
+            :borderRadius    4
+            :backgroundColor :white
+            :opacity         (if is-active 1.0 0.5)}}])
+
+(defn- indicator-list
+  [n current-index]
+  [rn/view
+   {:style {:display         :flex
+            :flex-direction  :row
+            :align-items     :center
+            :justify-content :center
+            :gap             8}}
+   (for [i (range n)]
+     ^{:key i} [indicator (= current-index i)])])
+
+(defn- render-item
+  [_]
+  (let [width (rf/sub [:dimensions/window-width])]
+    (reagent/create-class
+     {:should-component-update
+      (fn [_ [_ old-item] [_ new-item]]
+        (not= old-item new-item))
```

Comment:
this is required to prevent  `wallet-qr-code-item` component being re-rendered in the flatlist when the value of the `current-index` atom was updated.